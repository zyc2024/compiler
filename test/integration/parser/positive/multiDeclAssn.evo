

type ColorPoint = {
    int v;
}

type Color = {
    ColorPoint[] color;
}


int, bool, Color configData() {
    return 
        55,
        false,
        Color(
            color=[ColorPoint(v=255), ColorPoint(v=255), ColorPoint(v=255)]
        );
    
}

int, int, int getTriplet() {
    return 1,2,88;
}

int main() {
    int x, y, z;
    x, y, z = getTriplet();
    bool n2, n6, n, nn;
    Color c;
    x, n, c = configData();
}