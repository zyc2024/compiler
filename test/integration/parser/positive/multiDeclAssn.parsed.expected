(
    ()
    (
        (ColorPoint ((v int)))
        (Color ((color ([] ColorPoint))))
        (configData () (int bool Color) (
            (return 55 false (Color (
                ( = color (
                (ColorPoint ((= v 255)))
                (ColorPoint ((= v 255)))
                (ColorPoint ((= v 255))) )
            )))
        )))
        (getTriplet () (int int int) (
            (return 1 2 88)
        ))
        (main () (int) (
            (x y z int)
            (= (x y z) (getTriplet ()))
            (n2 n6 n nn bool)
            (c Color)
            (= (x n c) (configData ()))
        ))
    )
)